libraries/recorder.js

```diff
@@ -168,20 +172,29 @@ recorder.update = function() {
     }, 0);
 };
 
+/**
+ * @param {object} timeObject - object with keys
+ * representing sorted-increasing timestamps
+ * @param {number} targetTime
+ * @param {object|undefined} checkpoint
+ */
 recorder.replay = function(timeObject, targetTime, checkpoint) {
-    let objects = {};
-    let currentTime = 0;
-    if (checkpoint) {
+    let times = Object.keys(timeObject).map(t => parseInt(t));
+
+    let objects = JSON.parse(JSON.stringify(timeObject[times[0]]));
```

Comment:
Is stringify -> parse need for the undefined/null issue or is there other reason?

Reply:
ah, right.